packages/aws-cdk-lib/aws-lambda-nodejs/lib/function.ts

```diff
@@ -94,27 +109,41 @@ export class NodejsFunction extends lambda.Function {
       throw new Error('Only `NODEJS` runtimes are supported.');
     }
 
-    // Entry and defaults
-    const entry = path.resolve(findEntry(id, props.entry));
-    const handler = props.handler ?? 'handler';
-    const architecture = props.architecture ?? Architecture.X86_64;
-    const depsLockFilePath = findLockFile(props.depsLockFilePath);
-    const projectRoot = props.projectRoot ?? path.dirname(depsLockFilePath);
     const runtime = getRuntime(scope, props);
 
-    super(scope, id, {
-      ...props,
-      runtime,
-      code: Bundling.bundle(scope, {
-        ...props.bundling ?? {},
-        entry,
+    if (props.code !== undefined) {
+      if (props.handler === undefined) {
+        throw new Error('Cannot determine handler when `code` property is specified. Use `handler` property to specify a handler.');
+      }
+
+      super(scope, id, {
+        ...props,
         runtime,
-        architecture,
-        depsLockFilePath,
-        projectRoot,
-      }),
-      handler: handler.indexOf('.') !== -1 ? `${handler}` : `index.${handler}`,
-    });
+        code: props.code,
+        handler: props.handler!,
+      });
+    } else {
```

Comment:
Can you help me understand why we need this new code block here? Why can't we just determine the handler in the same way we are doing it right now? Couldn't we just keep the existing code and do something where if `code` is supplied we use it, otherwise we do `Bundling.bundle`? So something like:
```ts
  // Entry and defaults
  const entry = path.resolve(findEntry(id, props.entry));
  const architecture = props.architecture ?? Architecture.X86_64;
  const depsLockFilePath = findLockFile(props.depsLockFilePath);
  const projectRoot = props.projectRoot ?? path.dirname(depsLockFilePath);
  const handler = props.handler ?? 'handler';

  super(scope, id, {
    ...props,
    runtime,
    code: props.code ?? Bundling.bundle(scope, {
      ...props.bundling ?? {},
      entry,
      runtime,
      architecture,
      depsLockFilePath,
      projectRoot,
    }),
    handler: handler.indexOf('.') !== -1 ? `${handler}` : `index.${handler}`,
  });
}
```

Reply:
That's what I originally had, but I think it's a worse customer experience. Because there's nothing forcing the customer to name the file containing the exported handler `index`. It's entirely up to the customer to decide how they want to name that file, if they're using the `Code.fromCustomCommand`. Therefore, I think this is a better customer experience, even if it's a few more lines of code

Reply:
I see where you're coming from but I'm not sure I agree. I think this construct has already established that the expected behavior is we will default the handler if it isn't supplied. Consider a customer that is using the nodejs function and they currently don't supply a handler and just use the default. Now, once this feature is released, they decide to use the new `code` property to run a custom build script. However, suddenly they're encountering errors since they don't set a handler. I recognize that it is trivial to just specify a handler, but it is a change from the expectation for the construct as a whole. Additionally, we generally try to avoid properties where we have to do a bunch of validation around if a property is set then another one must also be set. We consider it a bit of an anti-pattern. Our VPC construct is a more extreme example of this, but it shows where this can get to be hard to deal with when we have to start doing a ton of property validation - https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-ec2/lib/vpc.ts

Reply:
basically, to  avoid having an outage in production, if a customer is switching to using this Code.fromCustomCommand, the customer would need to know that their build file must output a file called index.js that exports a function called handler. That's not something that I think should be hidden from the customer. By requiring handler, we don't abstract away a detail that could potentially lead to an outage

Reply:
I agree that this is more complicated, but the alternative is to push this complexity onto the customer after they've deployed their lambda

Reply:
Happy to change my mind on this, and with the way you've explained it I agree. Doesn't make sense to let the customer fail during deploy time.